File: java/iroha_android/src/main/java/org/hyperledger/iroha/android/nexus/UaidManifestCountMode.java
package org.hyperledger.iroha.android.nexus;

/** Counting mode accepted and emitted by the UAID manifest inventory endpoint. */
public enum UaidManifestCountMode {
  /** Return a bounded observed count. */
  BOUNDED("bounded"),

  /** Return the exact filtered count. */
  EXACT("exact");

  private final String parameterValue;

  UaidManifestCountMode(final String parameterValue) {
    this.parameterValue = parameterValue;
  }

  /** Returns the exact lowercase query/response spelling used by Torii. */
  public String parameterValue() {
    return parameterValue;
  }
}

File: kotlin/core-jvm/src/main/java/org/hyperledger/iroha/sdk/nexus/UaidManifestCountMode.kt
package org.hyperledger.iroha.sdk.nexus

/** Counting mode accepted and emitted by the UAID manifest inventory endpoint. */
enum class UaidManifestCountMode(
    /** Exact lowercase query/response spelling used by Torii. */
    val parameterValue: String,
) {
    /** Return a bounded observed count. */
    BOUNDED("bounded"),

    /** Return the exact filtered count. */
    EXACT("exact"),
}
